pythonforandroid/build.py

```diff
@@ -650,22 +653,22 @@ def build_recipes(build_order, python_modules, ctx):
             recipe.postbuild_arch(arch)
 
     info_main('# Installing pure Python modules')
-    run_pymodules_install(ctx, python_modules)
+    run_pymodules_install(
+        ctx, python_modules, project_dir,
+        ignore_setup_py=ignore_project_setup_py
+    )
 
     return
 
 
-def run_pymodules_install(ctx, modules):
+def run_pymodules_install(ctx, modules, project_dir, ignore_setup_py=False):
     modules = list(filter(ctx.not_has_package, modules))
```

Comment:
This method starts to be pretty big, do you think that would make sense to split it a bit more by concern or that's too painful?

Reply:
I vaguely recall that John Carmack wrote a nice article why splitting up big functions without actually making use of the inter-dependencies can actually be harmful. (because you get hidden dependencies between the split up functions increasing the risk later to mess things up when changing stuff, and the code flow is less obvious for no benefit)

I agree with this sentiment, and don't think there is any use in splitting it up just because it's long. However, if you think one part in there might be also needed to be called from somewhere else then that sounds like a good idea to me

Reply:
Well sometimes when the split by concern makes obvious sense. And sometimes it also simplifies unit testing it.
But it was just a suggestion, I didn't dig into it myself and I will not for now

Reply:
makes sense. for what it's worth, this function is the core install-the-whole-bunch place so I don't think it is very useful to be unit tested. (it's mainly relevant for the integration test where the non-split up code doesn't make a difference) so good point @ unit tests, but IMHO this should be ok